### ISSUE TITLE
Accessibility: product gallery full screen status not announced to screen reader users

### ISSUE THREAD

**Issue by amberhinds** [woothemes/woocommerce#43604]
### Prerequisites

- [X] I have carried out troubleshooting steps and I believe I have found a bug.
- [X] I have searched for similar bugs in both open and closed issues and cannot find a duplicate.

### Describe the bug

When toggling the gallery on the product single full screen there is a visual success message indicating that the gallery is now full screen:
<img width="948" alt="Full screen success message" src="https://github.com/woocommerce/woocommerce/assets/13153456/42e9ada7-c1fc-4db2-b75d-8ef385241026">

This message is not announced to screen reader users.

### Expected behavior

Status messages should be announced to screen readers.

Use [aria-live regions](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Live_Regions) to ensure screen reader announce status messages.

### Actual behavior

No message is head. Message is visual only.

### Steps to reproduce

1. Turn on a screen reader such as NVDA or VoiceOver.
2. Go to a single product and open the gallery.
3. Using your keyboard, click the full screen button.
4. Observe that a visual message is present but nothing is announced by the screen reader.

### WordPress Environment

N/A

### Isolating the problem

- [X] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [X] This bug happens with a default WordPress theme active, or [Storefront](https://woo.com/products/storefront/).
- [X] I can reproduce this bug consistently using the steps above.

**Comment by dinhtungdu** [comment]
Thanks for the report! I moved this to our backlog.

@thealexandrelara we should fix this issue in both classic gallery and new gallery block.

**Comment by thealexandrelara** [comment]
Thank you for reporting this, Amber. 

Also, thank you for the ping, Tung! I just created an issue to handle that for the new Product Gallery block.

**Comment by roykho** [comment]
Closed via https://github.com/woocommerce/woocommerce/pull/44332

**Comment by thealexandrelara** [comment]
@roykho  I'm going to re-open this issue because the changes on https://github.com/woocommerce/woocommerce/pull/44332 only fix the issue for the Product Gallery block. This reported issue is related to the Product Image Gallery block, which is a different block.

**Comment by amesplant** [comment]
I am looking into this.

**Comment by amesplant** [comment]
@amberhinds could I get a little bit more info on the issue that you are facing. When I visit an item in the [shop](https://shop.equalizedigital.com/product/make-wordpress-accessible-dark-mode-t-shirt/), using your shop as an example, my understanding is that when you enlarge (using the magnify icon), and then from there select the full screen option, that you see a dialog as in your report. I don't see such dialog in Safari or Chrome (Mac), and am wondering if I am not understanding the issue.

**Visiting Product**
![image](https://github.com/user-attachments/assets/e05d073a-178f-4853-92ab-c3217dbaa3c6)

**Selecting Full Screen**
![image](https://github.com/user-attachments/assets/fdf00e57-29b6-4d2e-bab0-b0733bf107ca)

**This is what appears**
![image](https://github.com/user-attachments/assets/f6cc03de-4c8a-4fea-b9ec-9873d4afaa8c)

**Comment by amberhinds** [comment]
Hi @amesplant, sorry I didn't specify a browser in the report. That screenshot was from Firefox.

If we can determine that the message is browser-specific—i.e., it is announced in other browsers but not in Firefox because the fullscreen mode uses browser-specific technology—then this issue may be deprioritized as a browser bug rather than a WooCommerce issue.

**Comment by amesplant** [comment]
@amberhinds I think that it is worth putting in functionality that announces when a screen enters into full size. I will spend a little time looking into the best way to accomplish this.

**Comment by amberhinds** [comment]
It would be most ideal if it was announced. Thanks!

**Comment by amesplant** [comment]
@amberhinds, again thanks for reporting the issue. I spent some time digging into this. 

I was able to inject some JavaScript to announce the Full Screen mode, but it gets interrupted by each browser's announcement, creating a worse situation for screen readers. And so while the visual message you caputred is not announced perse, a screen reader (testing with Mac OS VoiceOver) does seem to announce once in Fullscreen mode.

**Chrome**
![image](https://github.com/user-attachments/assets/89ce86a4-25ad-4754-b1d9-3f5a45cd3775)

**Safari**
![image](https://github.com/user-attachments/assets/338dde2b-075c-4217-a697-002d6c9a96a8)

**Firefox**
![image](https://github.com/user-attachments/assets/2c1d6e84-dfdb-45e2-9d9c-767162107a16)

**Comment by vikrampm1** [comment]
@amberhinds as per the comment above, there are no fixes needed on this issue. Let us know if you have any further questions or concerns, otherwise we suggest closing this issue.

**Comment by amberhinds** [comment]
I think we can close the issue.
